content/purgatorio/launch-and-mantain-the-system/monitoring-usage-and-behavior.md

```diff
@@ -1,5 +1,142 @@
+
+    - Consider whether training a model w/ out-of-domain detector (OOD) makes sense or not.
+
+    See: 
+    - [How to determine the confidence of a neural network prediction?](https://stats.stackexchange.com/questions/247551/how-to-determine-the-confidence-of-a-neural-network-prediction)
+
+- **Human evaluation:**
+
+    - Not really going to be avoidable, but you can use a lot of the above to narrow down what people look at. Have people look at the examples which are most alarming, based on the checks you have in place. Over time, we can better calibrate what alerts are of concern.
+
+    - A related option is to use a calibrated model of confidence/estimated error rate to inform what people look at. If you pick up the lowest 1% confidence and the internal estimated error rate is 5%, then if you audit that 1% and see > 5% error rate, that could be a concern. 
+
+    See: 
+    - [Comparing Automatic and Human Evaluation of Local Explanations for Text Classification](https://www.aclweb.org/anthology/N18-1097.pdf)
+
+
+## Monitoring the Usage
+
+If monitoring the behavior of the model can be technically hard, you should also be sure that your users are leveraging the model in the correct way. 
+
+With "users" we can refer to _everything that consumes the output of the ML model_, it can be a system, a human, or an ensemble of systems and humans.
+
+::: tip 
+If you're serving your model through an API (recommended way), you can refer to the API monitoring best practices in general (not specific for Machine Learning).
+:::
+
+See: 
+
+- [What's the best way to monitor your REST API?](https://stackoverflow.com/questions/28760946/whats-the-best-way-to-monitor-your-rest-api)
+- [Measure Anything, Measure Everything](https://codeascraft.com/2011/02/15/measure-anything-measure-everything/)
+
+The major issue you can encounter when dealing with people is that they choose to not use the ML model. 
+
+This can happen for a variety of reasons, maybe they don't have _confidence enough_ in the system, or they don't understand _how to_ use it.
+
+Take a look at this awesome guide from Google's engineers:
+
+> The People + AI Guidebook was written to help user experience (UX) professionals and product managers follow a human-centered approach to AI.
+
+- [People + AI Guidebook](https://pair.withgoogle.com/guidebook)
+
+This detailed resources can get you started about the following topics: 
+
+- User Needs + Defining Success
+- Data Collection + Evaluation
+- Mental Models
+- Explainability + Trust
+- Feedback + Control
+- Errors + Graceful Failure
+
+## Conclusions
+
+After walking through the resources listed here, you should be comfortable with the challenges and caveats of monitoring your Machine Learning model in production. 
+
+As you've seen, there are both technical challenges (data drift, input data check, output data check) and _"human-related"_ challenges. In particular, Google's People + AI Guidebook will show you most of the human-related ones.
```

Comment:
You want to put a footnote here which says:

- all of the above isn't a silver bullet
- use only those things that work for you and are applicable in your use-cases
- don't' literary follow any of the ideas, try to mould it to your needs but aim for the highest level of standards
- there may be others that are not listed here try them out and see how they work for you

_(and points like the above so you cover yourself from looking like a-list-that-solves-every-problem-for-all)_

Reply:
nice points, thanks